Varnish support

hshar7 opened the pull request:
I believe its ready to be merged. Please do one final CR

Commit:
Added Varnish support

Commit:
Added Varnish support

Commit:
Changed script to look at /etc dir instead of /vagrant/guest/etc/

Commit:
Merge remote-tracking branch 'origin/Guest-Configs' into varnish_support

Commit:
Merge origin/2.0 into varnish_support.

- Conflicts: configure_environment.sh

Commit:
Updated README

- To include information about Varnish.

hshar7 commented:
> Please update README.md and CHANGELOG.md
README updated. Not sure if can add to changelog since 2.0 was already published.

> Variable can be used for full path to magento2.conf
I believe it is easier to see the sed command if it wasn't in a variable. Can change though.

> It is better to introduce new script under scripts/guest called m-services-restart and create the same script on host (like is done for m-reinstall)
This will be separate task as we agreed.

paliarush commented on scripts/provision/configure_environment.sh:
```diff
@@ -29,6 +29,15 @@ a2ensite magento2.conf
 # Disable default virtual host
 sudo a2dissite 000-default
 
+# Copy varnish vcl file
```
This should be moved to https://github.com/paliarush/magento2-vagrant-for-developers/blob/2.0/scripts/provision/configure_environment_recurring.sh and probably adopted. We need to make sure that all existing installations will work correctly after pulling latest changes.

Commit:
Varnish Support

- Works now by invoking vagrant reload or by running "m-varnish -c enable" command.

Commit:
Merge remote-tracking branch 'remotes/origin/2.0' into varnish_support

Commit:
Varnish Support

- Changed name.
- Made script executable.
- Cache is cleared now after restarting services.

Commit:
Merge remote-tracking branch 'origin/2.0' into varnish_support

Commit:
Merge remote-tracking branch 'remotes/origin/2.0' into varnish_support

Commit:
Varnish Support

- Changed the way the command works.
- m-varnish command now accounts for backwards compatibility.
- m-varnish command now works on guest.

paliarush commented on etc/config.yaml.dist:
```diff
@@ -1,26 +1,26 @@
 repository_url:
-  # [Changeable: init-project -c]
+  # [To apply changes: init-project -c]
```
Actually should be `-fc`, `-c` by itself will not do anything at the moment. Maybe it make sense to change current behavior, but it might be tricky on Windows since it does not suppor NFS.

Commit:
Install varnish if it is not installed automatically on vagrant reload

Commit:
Install varnish if it is not installed automatically on vagrant reload

paliarush commented:
@hshar7 Thank you for PR with useful feature
